File: Week_03/169/majority_element.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counter = {}
        half_len = len(nums) // 2
        for n in nums:
            if n in nums:
                counter[n] += 1
            else:
                counter[n] = 1
            if counter[n] > half_len:
                return n
        return -1

File: Week_03/169/majority_element.go
package main

func majorityElement(nums []int) int {
	counter := make(map[int]int, 0)
	for _, n := range nums {
		counter[n]++
	}
	majorKey, majorValue := 0, 0
	for k, v := range counter {
		if v > majorValue {
			majorKey, majorValue = k, v
		}
	}
	return majorKey
}
